### PR TITLE
Remove branches to let tagged commits through

### PR DESCRIPTION
**Pull request by davidnsimplifi**
The travis builds for this are running, but the tagged commit does not appear to be pushing to hex, instead it gets "Skipping upload". It looks like this is because it's looking at the commit, but not when it's being tagged.

On other travis configs, we don't have that branches portion, which I believe could help with this problem. Not sure if there'd be other reasons to keep this, there is another possible option of using a deploy config in travis, but I'm not sure if that would work through the branches.

JIRA URL: https://simplifi.atlassian.net/browse/INT-8153

### CODE REVIEW

**Review by LiruMouse — APPROVED** [review]

**Comment by nathanmonteleone** [comment]
Do we not use `deploy:` blocks in the .travis.yml for any of our elixir libraries?  Is there a reason for that?  I ask because generally they're a bit easier to maintain than writing script logic to control when we upload.

**Comment by nathanmonteleone** [comment]
Oh, I found an example!

This is how I think we should be doing it, it's a lot cleaner: https://github.com/simplifi/ex_sifi_params/blob/33c14e85cc96ea51af61f0b2676fe7b0e489e0e5/.travis.yml#L25

**Comment by davidnsimplifi** [comment]
@nathanmonteleone Thanks, let me know what you think of this.

File: .travis.yml [hunk 1]
```diff
@@ -27,5 +27,10 @@ install:
...
-after_success:
-  - .travis/upload.sh
\ No newline at end of file
+deploy:
+  skip_cleanup: true
+  provider: script
+  script: mix hex.publish --yes
+  on:
+    tags: true
+    elixir: '1.13.4'
```
Comment (LiruMouse) [inline .travis.yml:36]: Does the elixir need to be restated, since it's mentioned above?

Comment (davidnsimplifi) [reply .travis.yml:36]: It looks like that's to limit the deploy to a specific elixir version out of the list: https://docs.travis-ci.com/user/deployment/#conditional-releases-with-on

I suppose it'd run on both if that was removed, although not sure what'd that do to hex, in terms of if one would get an error, or if it'd let it through.

Comment (LiruMouse) [reply .travis.yml:36]: Oh, ignore me, I thought this one had a single elixir in use.

**Comment by nathanmonteleone** [comment]
@davidnsimplifi LGTM!

**Review by nathanmonteleone — APPROVED** [review]

**Review by nick-gingerella — APPROVED** [review]
As someone who has been bitten hard by this "skipping upload" jank once upon a time, I'm grateful for any attempts to fix/avoid it in the future. Thanks David!

**Pull request merged**